geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/SafeTopologicalFilterVisitor.scala

```diff
@@ -12,16 +12,68 @@ import org.geotools.filter.visitor.DuplicatingFilterVisitor
 import org.locationtech.geomesa.filter.FilterHelper
 import org.opengis.feature.simple.SimpleFeatureType
 import org.opengis.filter.spatial._
+import org.opengis.filter.{And, Filter, Or}
 
-trait SafeTopologicalFilterVisitor extends DuplicatingFilterVisitor {
+import scala.collection.JavaConversions._
+
+/**
+  * Updates filters to handle IDL, dwithin units, and to remove filters that aren't meaningful
+  */
+trait SafeTopologicalFilterVisitor extends DuplicatingFilterVisitorAccessor {
+
+  import FilterHelper.{isFilterWholeWorld, rewriteDwithin, visitBBOX, visitBinarySpatialOp}
 
   def sft: SimpleFeatureType
 
-  override def visit(dw: DWithin, data: AnyRef) = FilterHelper.rewriteDwithin(dw)
-  override def visit(op: BBOX, data: AnyRef) = FilterHelper.visitBBOX(op, sft)
-  override def visit(op: Within, data: AnyRef) = FilterHelper.visitBinarySpatialOp(op, sft)
-  override def visit(op: Intersects, data: AnyRef) = FilterHelper.visitBinarySpatialOp(op, sft)
-  override def visit(op: Overlaps, data: AnyRef) = FilterHelper.visitBinarySpatialOp(op, sft)
+  override def visit(f: Or, data: AnyRef): AnyRef = {
+    val newChildren = f.getChildren.map(_.accept(this, data).asInstanceOf[Filter])
+    // INCLUDE OR foo == INCLUDE
+    if (newChildren.exists(includeEquivalent)) {
+      Filter.INCLUDE
+    } else {
```

Comment:
I'm curious why we don't have the

```scala
} else if (newChildren.length == 1) {
  newChildren.head
```

section as we do for the ```And``` filter in the next method.  Is it unnecessary here?